src/findReplace.js

```diff
@@ -219,147 +220,43 @@ export default class FindReplace extends Plugin {
-		const editor = this.editor;
-		if ( marker ) {
-			editor.model.change( writer => {
-				this._removeCurrentSearchMarker( writer );
-				this.currentSearchMarker = writer.addMarker( CURRENT_SEARCH_MARKER,
-					{ range: marker.getRange(), usingOperation: false } );
-			} );
-			const viewRange = editor.editing.mapper.toViewRange( marker.getRange() );
-			const domRange = editor.editing.view.domConverter.viewRangeToDom( viewRange );
-			scrollViewportToShowTarget( { target: domRange, viewportOffset: 130 } );
-		}
-	}
-
 	_resetStatus() {
 		this.findInput.infoText = undefined;
 		this.replaceInput.infoText = undefined;
-		this.currentSearchIndex = 0;
-		const model = this.editor.model;
-		model.change( writer => {
-			for ( const searchMarker of model.markers.getMarkersGroup( SEARCH_MARKER ) ) {
-				writer.removeMarker( searchMarker );
-			}
-			this._removeCurrentSearchMarker( writer );
-		} );
+		removeSearchMarkers( this.editor.model );
 	}
 
-	_removeCurrentSearchMarker( writer ) {
-		if ( this.currentSearchMarker ) {
-			writer.removeMarker( this.currentSearchMarker );
-			this.currentSearchMarker = undefined;
-		}
+	_replace( findField, replaceField,replaceAll=false ) {
+		const findText = findField.fieldView.element.value;
+		const replaceText = replaceField.fieldView.element.value;
+		const {currentIndex,total} = this.editor.execute('findReplace',{
+			findText,
+			replaceText,
+			replaceAll
+		})
+		this._updateFindInfo(findField,currentIndex,total)
 	}
 
-	_replace( findField, replaceField ) {
-		const model = this.editor.model;
-		const markers = Array.from( model.markers.getMarkersGroup( SEARCH_MARKER ) );
-		const sameSearch = this._isSameSearch( findField, markers );
-		const currentMarker = sameSearch ? markers[ this.currentSearchIndex ] : this._find( findField, 1 );
-		const replaceBy = replaceField.fieldView.element.value;
-		if ( currentMarker && currentMarker.getRange ) {
-			model.change( writer => {
-				model.insertContent( writer.createText( replaceBy ), currentMarker.getRange() );
-				writer.removeMarker( currentMarker );
-				this._removeCurrentSearchMarker( writer );
-			} );
-			// refresh the items...
-			this._find( findField, 0 );
-		}
+	_replaceAll( findField, replaceField ) {
+		this._replace(findField,replaceField,true)
 	}
 
-	_replaceAll( findField, replaceField ) {
-		const model = this.editor.model;
+	_updateFindInfo(findField,currentIndex=0,total=0){
 		const t = this.editor.t;
-		// fires the find operation to make sure the search is loaded before replace
-		this._find( findField, 1 );
-		const replaceBy = replaceField.fieldView.element.value;
-		model.change( writer => {
-			const markers = model.markers.getMarkersGroup( SEARCH_MARKER );
-			let size = 0;
-			for ( const marker of markers ) {
-				model.insertContent( writer.createText( replaceBy ), marker.getRange() );
-				size++;
-			}
-			this._resetStatus();
-			replaceField.infoText = t( 'Replaced ' ) + size + t( ' times' );
```

Comment:
I think this was lost in the new version. Without it, the user get no feedback. the replace method could return total number of replacements (in the normal use, would be always 1, but in the replaceAll version, it should be variable)